vignettes/special-files.Rmd

```diff
@@ -0,0 +1,77 @@
+---
+title: "Special files"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Special files}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup}
+#| include: false
+library(testthat)
+```
+
+This vignette describes the various special files that testthat works with: test, helper, and setup files.
+
+## Test files
+
+These are bread and butter of testthat.
+Test files live in `tests/testthat`, start with either `test-` or `test_`, and end with `.r` or `.R`.
+We recommend organising your test files so that there's a one-to-one correspondence between the files in `R/` and the files in `tests/testthat/` so that (e.g.) `R/file-one.R` has a matching `tests/testthat/test-one.R`.
+This correspondence is maintained by functions like `usethis::use_r()` and `usethis::use_test()` and by respected by functions like `devtools::test_active_file()` and `devtools::test_coverage_active_file()`.
+
+Test files are executed in alphabetical order, but you should strive to avoid dependencies between test files.
```

Comment:
Say something about "ideally test files can run in parallel without interfering with each other"?